src/docfx/config/GlobConfig.cs

```diff
@@ -47,32 +50,30 @@ public bool Match(string filePath)
         {
             Debug.Assert(!string.IsNullOrEmpty(filePath));
 
-            if (Exclude.Any(e => MatchItem(filePath, e)))
-                return false;
-            if (Include.Any(i => MatchItem(filePath, i)))
-                return true;
-
-            return false;
-        }
-
-        private bool MatchItem(string filePath, string pattern)
-        {
-            if (pattern == null)
-                return false;
-
             if (IsGlob)
             {
-                // TODO: optimize this
-                var options = PathUtility.IsCaseSensitive ? GlobMatcher.DefaultCaseSensitiveOptions : GlobMatcher.DefaultOptions;
-                var glob = new GlobMatcher(pattern, options);
-                return glob.Match(filePath);
+                return LazyInitializer.EnsureInitialized(
+                    ref _glob,
```

Comment:
>_glob [](start = 24, length = 5)

can it just be a local variable?

Reply:
This ensure `_glob` is only initialized once. the `CreateGlobMatcher` method is more expensive compare to the `match` function itself

Reply:
Thanks for the explanation